linera-core/src/client/mod.rs

```diff
@@ -1044,15 +1050,40 @@ where
             blob_ids: required_blob_ids,
         };
         let certificate = self
-            .communicate_chain_action(committee, submit_action, value)
+            .communicate_chain_action(&committee, submit_action, value)
             .await?;
         self.process_certificate(certificate.clone(), proposed_blobs)
             .await?;
-        if certificate.inner().is_confirmed() {
-            Ok(certificate.try_into().unwrap()) // shouldn't panic, we just checked.
+        let certificate = if certificate.inner().is_confirmed() {
+            certificate.try_into().unwrap() // shouldn't panic, we just checked.
         } else {
-            self.finalize_block(committee, certificate.into()).await
+            self.finalize_block(&committee, certificate.into()).await?
+        };
+
+        // Communicate the new certificate now.
+        let next_block_height = self.next_block_height();
+        self.communicate_chain_updates(
+            &committee,
+            self.chain_id,
+            next_block_height,
+            self.options.cross_chain_message_delivery,
+        )
+        .await?;
+        if let Ok(new_committee) = self.local_committee().await {
+            if new_committee != committee {
+                // If the configuration just changed, communicate to the new committee as well.
+                // (This is actually more important that updating the previous committee.)
+                let next_block_height = self.next_block_height();
+                self.communicate_chain_updates(
+                    &new_committee,
+                    self.chain_id,
+                    next_block_height,
+                    self.options.cross_chain_message_delivery,
+                )
+                .await?;
+            }
```

Comment:
Not in this PR but we can discuss moving things like that to a "daemon process" within the client. Something that would subscribe to these changes and communicate chain updates if necessary.

Reply:
Possibly. I guess the `ClientListener` could do this whenever it gets a `NewBlock` notification. :thinking:

Reply:
Sounds good, at first reading 👍

Reply:
But let's leave that change for later, too. Currently we don't know whether a listener is running at all. If it isn't, and this block was created due to a CLI command, then we should communicate the chain updates here before returning. Otherwise other users might not see our new block at all!

Reply:
Of course, not in this PR. Will you ticket it or should I do it?

Reply:
I think it might need some discussion with the team first.

The client is just currently not set up in a way that it's always running a chain listener, so this wouldn't work yet.

Reply:
I was imaginining that client needs to maintain some kind of listener/subscription with the chain/network to receive things like events, incoming messages, etc. and this could be added there.

Reply:
Yes, but only in "connected" mode, i.e. if it's running the node service (`linera service`) or in the browser. It currently still works without a running listener when executing a single CLI command: In that case it just synchronizes actively from the network and then tries to propose its block, and _then_, if it turns out it has to wait for a timeout, it might start a listener.

But I agree that we should change that, for simplicity, and because individual CLI commands are not something we want to optimize for. It's just that this will be a bigger change again, but I agree that we should do it.